sqlmesh/core/engine_adapter/snowflake.py

```diff
@@ -90,7 +90,10 @@ def query_factory() -> Query:
                 #
                 # The above issue has already been fixed upstream, but we keep the following
                 # line anyway in order to support a wider range of Snowflake versions.
-                self.cursor.execute(f'USE SCHEMA "{temp_table.db}"')
+                schema = f'"{temp_table.db}"'
+                if temp_table.catalog:
+                    schema = f'"{temp_table.catalog}".{schema}'
```

Comment:
The double quotes around `schema` were accidentally removed here

Reply:
The double quotes are already in the `schema` var. Here is the output from the snowflake log - `USE SCHEMA "SQLMESH_TEST2"."sqlmesh__EXAMPLE2"`

Reply:
sorry yep missed that